### DISCUSSION TITLE
New website

### DISCUSSION THREAD

**Discussion by FredrikAugust** [open-source-ideas/ideas#307] [General]
Several people have express a wish for a new/remade website. 

Features, implementation details, tech stack, etc can be discussed here.

**Comment by FredrikAugust** [comment]
Please see #59 and #208 for the previous discussions.

**Comment by fire17** [comment]
We can draw some requested features from the prev conversations, but I think a good start will be something basic and low to zero cost.

As we've discussed the main backbone should still remain this repo, and the data just needs to be represented better.

I seems possible to use the GitHub API on a GitHub-page thats perfect. cost nothing,  but the data is always be up to date with the repo.  https://open-source-ideas.github.io/
Just needs to be more modern looking, with some pagination/search/filter options

This could be the first milestone 
What do you guys think ?

**Reply by KaKi87** [reply]
I would understand this choice. Personally I like not to depend on centralized providers, so I prefer building and hosting everything myself, my favourite front-end stack being Vue JS bundled with Parcel.

**Reply by fire17** [reply]
Well you're very right, but in this case.. all the data is originaly on GitHub, so GitHub pages doesnt really add more dependency

Vue is amazing
Will look into Parcel :)

**Reply by FredrikAugust** [reply]
Personally, I would prefer relying on a centralised provider simply for the reliability and ease of use. I can cover the costs if there are any myself. 

Other than that, I don't have any strong preferences! Vue sounds good if that's what you're comfortable with

**Reply by KaKi87** [reply]
I am totally comfortable with Vue for front-end development, however I need a UI/UX designer to provide Adobe XD/Figma prototypes based on discussed specifications, because I don't do design myself.

Also I would suggest considering mobile-first design.

**Reply by fire17** [reply]
@KaKi87 (or anyone)
with your experience can make a simple draft maybe ? Something that can be practical for any feature you put , that can be later edited to to have better design and visuals ?

Even a base boilerplate would be very nice. The minimum things I guess are reading from the github api and displaying issues in either a main page that scrolles through the projects (title, desc, meta) or in a page of their own (which can be more detailed and display comments, etc). Also pagination and or search would be pleasant.

Can gh-pages host Vue sites ? If not I bet this sort of site will be just fine free on Heroku , or hosted cheaply by someone in the community. A github-page solution should be the quickest and simplest in my opinion but Vue is much more powerful and smart to use in the long run


> Also I would suggest considering mobile-first design.
Fully agree

**Reply by KaKi87** [reply]
> reading from the github api and displaying issues in either a main page that scrolles through the projects (title, desc, meta) or in a page of their own (which can be more detailed and display comments, etc). Also pagination and or search would be pleasant.

I can make a browser JS lib, but I won't make a UI, because any attempt at UI/UX (especially UX) design will be biased by the existing PoC.

> Can gh-pages host Vue sites ?

I have no idea.

> site will be just fine free on Heroku

I have no idea.

> hosted cheaply by someone in the community

I have a dedicated server which already hosts a dozen of services, I can add this one without any cost or issue.

**Reply by FredrikAugust** [reply]
I don't see why you would need to create a JS library. You can use [the official Github SDK for JavaScript](https://github.com/octokit/octokit.js#graphql-api-queries) to retrieve all the data from the Github API, and wrap it up nicely with Vue/React/Svelte/Elm/&c.

Vue "compiles" to a static website, so yes, it can be served by gh-pages. Unless you intend to have a custom backend, it should suffice.

> A github-page solution should be the quickest and simplest in my opinion but Vue is much more powerful and smart to use in the long run

I don't quite understand what you mean, @fire17. You can host the Vue app with gh-pages. Are you thinking of using markdown files to add pages to jekyll ([like this](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/adding-content-to-your-github-pages-site-using-jekyll))?

**Reply by fire17** [reply]
Cool! I wasn't sure that Vue and gh-pages are compatible, so I meant if not we can use Jekyll cause it's easy, but yea Vue is awesome,

If KaKi can make a simple site in Vue later it could be more refined.
Just need use the GitHub API for now and a touch of order (pagination or search or filter or categories etc)

**Reply by KaKi87** [reply]
> I don't see why you would need to create a JS library.

It is not *needed*, but I could do it just to says that it's done.

> You can use the official Github SDK for JavaScript to retrieve all the data

I know.

> If KaKi can make a simple site in Vue later it could be more refined.

On the contrary, I would prefer that a UI & UX designer creates mockups for the fully-featured website first, so that I could implement it entirely.

**Reply by FredrikAugust** [reply]
> On the contrary, I would prefer that a UI & UX designer creates mockups for the fully-featured website first, so that I could implement it entirely.

I agree with @fire17 here. I think working iteratively here is the way to go. Unless someone knows a designer who would be willing to create a complete design for the website for free, I think we're going to have to wait a long time for one to stumble by. Creating mockups for a fully-featured website would also require us to know all the features we would like.

If you could isolate one feature which would provide a benefit over simply using Github and perhaps implement a MVP for that, then it will be easier to work with that.

**Reply by KaKi87** [reply]
> Unless someone knows a designer who would be willing to create a complete design for the website for free

Well, I'm a web developer who would be willing to create a complete web app for free, provided that UI/UX mockups are done.

The solution you're describing is how FLOSS projects usually works, because it allows faster distribution, but it also discards UX, which I do not want to do. I'm sorry, but I am not willing to work that way.

**Reply by fire17** [reply]
Well I think what you are describing is very professional and just a level beyond the basic scope of the first milestone 

For later stages I'm sure XD/Figma will be fantastic to keep the UX as best as possible . We've discussed some before 
, i see a big vision here and many cool features that could be added. I'm mostly interested in network effects down the line, having a profile for myself I can manage, being able to star projects, following other people etc. 

Do you have to have actual mockup assets to start a simple site ? 
I'm just not familiar with your workflow.. but if you look at the current gh-page
https://open-source-ideas.github.io/
 you can see what it offers (id just use the base Vue layout as it's way more modern, and add basic pagination) just that could be one big leap from the current state

I'm assuming with your capabilities it's doable, but if you prefer differently , then maybe you know someone that can help you make these mockups . 

Just trying to keep this thing moving and incrementally make progress
Lemme know if what I'm saying makes any sense

Also as @FredrikAugust said, it's not currently clear what the fully featured site will offer or look like. We do need to change that. If I remembered correctly, you had a stackoverflow design in mind, while I like the network features. Since it's oriented around the same data, different views can represent it so we can always AB test and see what's best. I suggest opening a repo (or using existing) that describes in the readme possible features - arranged their priorities. Then we can go back through the conversations and copy them over to get a more defined UX design

**Comment by KOLANICH** [comment]
I don't think the website is needed at all, given that all the activity is done on GitHub.

**Reply by fire17** [reply]
GitHub is nice and all but there are some features missing, and a clear webpage will be much nicer to follow.
Please check out #208 has it has many features requested that cannot done here natively

**Reply by FredrikAugust** [reply]
I don't think it's _needed_, but if someone wants to create a website to supplement the Github page, then I don't see a problem with that.

**Reply by fire17** [reply]
Just to clarify, all the current activity can still be done regularly on GitHub as it is the backbone of the site, it's not a split platform, just a different way to view the same data

**Comment by fire17** [comment]
> Features, implementation details, tech stack, etc can be discussed here.

 @FredrikAugust do you think we can make a place where we can copy over, sort and stack the suggested features described in these threads and previous ones? so we can aggregate and prioritize them, which will help refine milestones and lead us to have clearer UI/UX designs

hopefully bringing all of us to the same page, where we can see what's already been done and what's next on the agenda

if you could please link here to some relevant readme or let us know your preferred method to achieve this 🙏

**Reply by FredrikAugust** [reply]
Yep. I can create a new repository for the new website and add you all as contributors. Would that work?

**Reply by FredrikAugust** [reply]
Bump @fire17

**Reply by fire17** [reply]
@FredrikAugust Yes please send details :)
